### DISCUSSION TITLE
Single file builds

### DISCUSSION THREAD

**Discussion by aklofas** [jekyll/jekyll#8937] [Ideas]
Hello,

What do the devs think about adding a command-line option to build specific files only (rather than entire source folder)? This will allow jekyll integration into eg. makefiles, gulp, etc. for more complex build pipelines.

example usage:
`jekyll build --source src/ --destination build/ --file index.html`

**Comment by ashmaroli** [comment]
Jekyll CLI currently has an option `--incremental` where only the *changed file and its dependencies* get built.
